frame/alliance/src/lib.rs

```diff
@@ -604,25 +617,53 @@ pub mod pallet {
 		}
 
 		/// Initialize the founders, fellows, and allies.
+		/// Resets members and removes all active proposals if alliance is already initialized.
 		///
-		/// This should only be called once, and must be called by the Root origin.
-		#[pallet::weight(T::WeightInfo::init_members(
+		/// Must be called by the Root origin.
+		/// To force reset the Alliance, witness data must be provided.
+		#[pallet::weight(T::WeightInfo::force_set_members(
 			T::MaxFounders::get(),
 			T::MaxFellows::get(),
-			T::MaxAllies::get()
+			T::MaxAllies::get(),
+			witness.proposals,
+			witness.members,
 		))]
-		pub fn init_members(
+		pub fn force_set_members(
 			origin: OriginFor<T>,
 			founders: Vec<T::AccountId>,
 			fellows: Vec<T::AccountId>,
 			allies: Vec<T::AccountId>,
+			witness: ForceSetWitness,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			// Cannot be called if the Alliance already has Founders or Fellows.
-			// TODO: Remove check and allow Root to set members at any time.
-			// https://github.com/paritytech/substrate/issues/11928
-			ensure!(!Self::is_initialized(), Error::<T, I>::AllianceAlreadyInitialized);
+			if Self::is_initialized() {
+				// Reset Alliance by removing all members.
+				// Veto and remove all active proposals to avoid any unexpected behavior from
+				// actionable items managed outside of the pallet. Items managed within the pallet,
+				// like `UnscrupulousWebsites`, are left for the new Alliance to clean up or keep.
+
+				let proposals = T::ProposalProvider::proposals();
+				ensure!(proposals.len() as u32 <= witness.proposals, Error::<T, I>::BadWitness);
+
+				let votable_members = Self::votable_members();
+				ensure!(votable_members.len() as u32 <= witness.members, Error::<T, I>::BadWitness);
+
+				proposals.into_iter().for_each(|hash| {
+					T::ProposalProvider::veto_proposal(hash);
+				});
+
+				T::MembershipChanged::change_members_sorted(&[], &votable_members, &[]);
```

Comment:
The extrinsic is currently ignoring the weight of these callbacks.  
There is not really an ideal solution besides giving the trait a weightinfo and then using it in the `pallet::weight` annotation. Something like (pseudo code):  
```rust
force_set_members.weight + veto_proposal.weight * witness.proposals + change_members_sorted.weight  * witness.members
```

Reply:
I understand its not ideal but wont the `force_set_members` benchmark include the weights and db reads/writes of `veto_proposal` and `change_members_sorted` if the actual `ProposalProvider` and `MembershipChanged` injected for benchmark?

if yes, I use `decode_len`s for witness checks and it might be good enough for the purpose of this PR.

Reply:
> veto_proposal and change_members_sorted if the actual ProposalProvider and MembershipChanged injected for benchmark?

Yes it will, but then you make the assumption that all blockchains will use exactly these as well. 
In that case the abstraction via a trait is void.  
In general you cannot make assumptions about the weight of a trait function.  

Anyway, its good for now so lets record it in a new issue.